storehaus-cassandra/src/main/scala/com/twitter/storehaus/cassandra/cql/cassandraTupleStores.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twitter.storehaus.cassandra.cql
+
+import com.datastax.driver.core.Row
+import com.twitter.concurrent.Spool
+import com.twitter.storehaus.{IterableStore, ReadableStore, QueryableStore, Store}
+import com.twitter.util.{Future, FutureTransformer, Return, Throw}
+import shapeless.ops.hlist.Tupler
+import shapeless.syntax.std.tuple._
+import shapeless.HList
+import shapeless._
+import com.twitter.storehaus.cassandra.cql.macrobug._
+
+/**
+ * AbstractCQLCassandraCompositeStore-wrapper for Tuples to HList. This makes it 
+ * easy to use this store with tuples, which can in turn be serialized easily.
+ */
+class CassandraTupleStore[RKT <: Product, CKT <: Product, V, RK <: HList, CK <: HList, RS <: HList, CS <: HList]
+		(val store: AbstractCQLCassandraCompositeStore[RK, CK, V, RS, CS], paramToPreventWritingDownTypes: (RKT, CKT))
+		(implicit ev1: HListerAux[RKT, RK],
```

Comment:
Using `Generic` would give you both directions in a single type class (and allow you to avoid all the `HListerAux` boilerplate). If you want evidence that you've got a tuple (as opposed to some other `Product`, there's an `IsTuple` type class).